docs/commerce/shipping_management/extend_shipping.md

```diff
@@ -0,0 +1,157 @@
+```
+
+In `translations/`, create a translations file `ibexa_shipping.en.yaml` that stores a name value for the custom shipping method type:
+
+``` yaml
+[[= include_file('code_samples/front/shop/shipping/translations/ibexa_shipping.en.yaml') =]]
+```
+
+Next, use the type factory to define an options form mapper class in `config/services.yaml`:
+
+``` yaml
+[[= include_file('code_samples/front/shop/shipping/config/packages/services.yaml', 0, 1) =]][[= include_file('code_samples/front/shop/shipping/config/packages/services.yaml', 8, 15) =]]
+```
+
+![Creating a shipping method of custom type](img/custom_shipping_method_type.png "Creating a shipping method of custom type")
+
+### Create options validator
+
+You might want to validate the data provided by the user against certain constraints. 
+Here, you create an options validator class that checks whether the user provided the `customer_identifier` value and dispatches an error when needed.
+
+Use the type factory to define a compound validator class in `config/services.yaml`:
+
+``` yaml
+[[= include_file('code_samples/front/shop/shipping/config/packages/services.yaml', 0, 1) =]][[= include_file('code_samples/front/shop/shipping/config/packages/services.yaml', 15, 22) =]]
+```
+
+Then, in `src/ShippingMethodType/`, create a `CustomerNotNullValidator.php` with the validator class:
+
+``` php
+[[= include_file('code_samples/front/shop/shipping/src/ShippingMethodType/CustomerNotNullValidator.php') =]]
+```
+
+Finally, register the validator as a service:
+
+``` yaml
+[[= include_file('code_samples/front/shop/shipping/config/packages/services.yaml', 0, 1) =]][[= include_file('code_samples/front/shop/shipping/config/packages/services.yaml', 22, 25) =]]
+```
+
+### Create storage converter
+
+Before form data can be stored in database tables, field values must be converted to a storage-specific format.
+Here, the storage converter converts the `customer_identifier` string value into the `customer_id` numerical value.
+
+In `src/ShippingMethodType/Storage`, create a `StorageConverter.php` file with the storage converter class:
+
+``` php
+[[= include_file('code_samples/front/shop/shipping/src/ShippingMethodType/Storage/StorageConverter.php') =]]
+```
+
+Then, register the storage converter as a service:
+
+``` yaml
+[[= include_file('code_samples/front/shop/shipping/config/packages/services.yaml', 0, 1) =]][[= include_file('code_samples/front/shop/shipping/config/packages/services.yaml', 25, 28) =]]
+```
+
+#### Storage definition 
+
+Now, create a storage definition class and a corresponding schema.
+The table stores information specific for the custom shipping method type.
+
+!!! note "Create table"
+
+    Before you can proceed, in your database, create a table that has columns present in the storage definition, for example:
+
+    `CREATE TABLE ibexa_shipping_method_region_custom(id int, customer_id text, shipping_method_region_id int);`
```

Comment:
Actually, the `id` column is always `NULL`.
I think that the column `id` seem unused because it has to fill itself automatically:
```suggestion
    `CREATE TABLE ibexa_shipping_method_region_custom(id int auto_increment primary key, customer_id text, shipping_method_region_id int);`
```